common/djangoapps/xblock_django/user_service.py

```diff
@@ -35,6 +35,13 @@ def get_current_user(self):
         """
         return self._convert_django_user_to_xblock_user(self._django_user)
 
+    def get_external_user_id(self, type_name):
+        """
+        Returns an external user id of the given type.
+        """
+        external_id, _ = ExternalId.add_new_user_id(self._django_user, type_name, create_type=True)
```

Comment:
Why do we not need to generate types dynamically? How many types will be generated dynamically?

The original intent of this table was to generate external IDs for learners based on a known type like coaching.  I am concerned that this will result in types being created without a clear definition or meaning. From that, it will be difficult to know what to use.

Could you expand on the following?

- How will these auto-generated types be used?
- How will they be generated?
- How will we know the type when we want to use the ID?

My concern here is that depending on how this is used it could have an impact on performance very quickly.